plugins/modules/aci_route_context_scope.py

```diff
@@ -0,0 +1,327 @@
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        l3out=dict(type="str", aliases=["l3out_name"]),
+        profile=dict(type="str", aliases=["profile_name", "route_control_profile"]),
+        context=dict(type="str", aliases=["name"]),
+        description=dict(type="str", aliases=["descr"]),
+        set_action_rule_profile=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["l3out", "tenant", "profile", "context"]],
+            ["state", "present", ["l3out", "tenant", "profile", "context", "set_action_rule_profile"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    l3out = module.params.get("l3out")
+    state = module.params.get("state")
+    tenant = module.params.get("tenant")
+    profile = module.params.get("profile")
+    context = module.params.get("context")
+    description = module.params.get("description")
+    set_action_rule_profile = module.params.get("set_action_rule_profile")
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="l3extOut",
+            aci_rn="out-{0}".format(l3out),
+            module_object=l3out,
+            target_filter={"name": l3out},
+        ),
+        subclass_2=dict(
+            aci_class="rtctrlProfile",
+            aci_rn="prof-{0}".format(profile),
+            module_object=profile,
+            target_filter={"name": profile},
+        ),
+        subclass_3=dict(
+            aci_class="rtctrlCtxP",
+            aci_rn="ctx-{0}".format(context),
+            module_object=context,
+            target_filter={"name": context},
+        ),
+        subclass_4=dict(
+            aci_class="rtctrlScope",
+            aci_rn="/scp",
+            module_object="",
+            target_filter={"name": ""},
```

Comment:
The name for the scope is empty by default. I could set the module_object and target_filter to None, but I think that would break the ability to do a query all.